src/picklescan/scanner.py

```diff
@@ -307,12 +327,35 @@ def scan_pickle_bytes(data: IO[bytes], file_id, multiple_pickles=True) -> ScanRe
     return _build_scan_result_from_raw_globals(raw_globals, file_id)
 
 
+# XXX: it appears there is not way to get the byte stream for a given file within the 7z archive and thus forcing us to unzip to disk before scanning
+def scan_7z_bytes(data: IO[bytes], file_id) -> ScanResult:
+    result = ScanResult([])
+
+    archive = py7zr.SevenZipFile(data, mode="r")
```

Comment:
Skimming through the py7zr doc it looks like the archive should be accessed within a `with` block.